libdebug/cffi/ptrace_cffi_source.c

```diff
@@ -501,6 +501,96 @@ int cont_all_and_set_bps(struct global_state *state, int pid)
     return status;
 }
 
+int prepare_thread_for_run(struct global_state *state, int pid, int tid)
```

Comment:
So, this function is identical to `prepare_for_run` but the operations in the second while are performed for the selected thread and not for all threads. Can we have a function called by both `prepare_thread_for_run` and `prepare_for_run` to avoid copy-paste of code?